public/main.js

```diff
@@ -16,63 +16,142 @@ function convertDate(timestamp) {
 const $expenseTable = document.querySelector('#expense-table')
 const $expenseBody = document.querySelector('.expense-body')
 
-function renderExpenseData(data) {
+function renderTransactionData(data) {
   const $container = document.createElement('tr')
   const $categoryData = document.createElement('td')
-  const $expenseData = document.createElement('td')
+  const $transactionData = document.createElement('td')
   const $dateData = document.createElement('td')
   const $amountData = document.createElement('td')
 
   $categoryData.textContent = data.category
-  $expenseData.textContent = data.item
+  $transactionData.textContent = data.item
   $dateData.textContent = convertDate(data.transaction_date)
-  $amountData.textContent = '$ ' + data.amount
+  if (data.type === 'expense') {
+    $amountData.textContent = '- $ ' + data.amount
+  }
+  else if (data.type === 'income') {
+    $amountData.textContent = '+ $ ' + data.amount
+  }
 
   $container.appendChild($categoryData)
-  $container.appendChild($expenseData)
+  $container.appendChild($transactionData)
   $container.appendChild($dateData)
   $container.appendChild($amountData)
 
   return $container
 }
 
-function totalExpense(data) {
+
+function organizeAmounts(data) {
+  const amounts = { incomeAmounts: [], expenseAmounts: [] }
+  for (let i = 0; i < data.length; i++) {
+    if (data[i].type === 'income') {
+      const amount = Number(data[i].amount)
+      amounts.incomeAmounts.push(amount)
+    }
+    else if (data[i].type === 'expense') {
+      const amount = Number(data[i].amount)
+      amounts.expenseAmounts.push(amount)
+    }
+  }
+  return amounts
+}
+
+
+function sum(data) {
   let total = 0
   for (let i = 0; i < data.length; i++) {
-    const amount = Number(data[i].amount)
+    const amount = data[i]
     total += amount
   }
   return total
 }
 
 
+function netTotal(income, expense) {
+  const total = (income - expense)
+  return total
+}
+
+
+const $transactions = document.querySelector('#transactions')
+
+function appendTotalTrans(data) {
+  const total = data.length
+  $transactions.textContent = total
+}
+
+const $totalIncome = document.querySelector('#income-total')
+
+function appendTotalIncome(total) {
+  $totalIncome.textContent = '$ ' + total
+}
+
 const $totalExpenses = document.querySelector('#expense-total')
 
 function appendTotalExpense(total) {
   $totalExpenses.textContent = '$ ' + total
 }
 
-const $transactions = document.querySelector('#transactions')
+const $netTotal = document.querySelector('#net-total')
 
-function appendTotalNumber(data) {
-  const total = data.length
-  $transactions.textContent = total
+function appendNetTotal(total) {
+  if (total > 0) {
+    $netTotal.textContent = '$ ' + total
+  }
+  else if (total < 0) {
+    $netTotal.textContent = '- $ ' + (total * -1)
+  }
+  else if (total = 0) {
+    $netTotal.textContent = '$ ' + total
```

Comment:
both the first and the third case are the same, so maybe you should just check `if (total >= 0)`